lms/services/application_instance.py

```diff
@@ -33,5 +33,8 @@ def get(self, consumer_key=None) -> ApplicationInstance:
 
 
 def factory(_context, request):
-    consumer_key = request.lti_user.oauth_consumer_key if request.lti_user else None
+    # TODO this is silly, we are picking now the consumer_key from the actual AI
+    consumer_key = (
+        request.lti_user.application_instance.consumer_key if request.lti_user else None
+    )
```

Comment:
I'd say if nothing is wrong with the approach of this PR this param should be removed.